src/diffusers/utils/hub_utils.py

```diff
@@ -144,6 +144,32 @@ def create_model_card(args, model_name):
     model_card.save(card_path)
 
 
+# Taken from `transformers`
+def create_and_tag_model_card(repo_id: str, token: Optional[str] = None, is_pipeline=False):
+    """
+    Creates or loads an existing model card and tags it with the `library_name`.
+
+    Args:
+        repo_id (`str`):
+            The repo_id where to look for the model card.
+        token (`str`, *optional*):
+            Authentication token, obtained with `huggingface_hub.HfApi.login` method. Will default to the stored token.
+        is_pipeline (`bool`, *optional*):
+            Boolean to indicate if we're adding tag to a [`DiffusionPipeline`].
+    """
+    try:
+        # Check if the model card is present on the remote repo
+        model_card = ModelCard.load(repo_id, token=token, ignore_metadata_errors=False)
+    except EntryNotFoundError:
+        # Otherwise create a simple model card from template
+        component = "pipeline" if is_pipeline else "model"
+        model_description = f"This is the model card of a 🧨 diffusers {component} that has been pushed on the Hub. This model card has been automatically generated."
+        card_data = ModelCardData(library_name="diffusers")
+        model_card = ModelCard.from_template(card_data, model_description=model_description)
```

Comment:
Will prefer templatizing the model card in a separate PR. 

Does 5297ad46 work for you?

Reply:
> Does https://github.com/huggingface/diffusers/commit/5297ad46f747ca462585c53c1d67872953e3b62e work for you?

The logic is okay but I would tend to do something like this:

```py
# Load model card or create a new one
try:
    model_card = ModelCard.load(repo_id, token=token)
except EntryNotFoundError:
    component = "pipeline" if is_pipeline else "model"
    model_description = f"This is the model card of a 🧨 diffusers {component} that has been pushed on the Hub. This model card has been automatically generated."
    model_card = ModelCard.from_template(ModelCardData(), model_description=model_description)

# Update metadata
if model_card.data.library_name is None:
    model_card.data.library_name = "diffusers"

# Return
return model_card
```

to really separate the part where the model card is loaded/generated and the part we populate new fields. The advantage is that if you need to add more metadata in the future it would be at the same consistent place.

Reply:
Makes sense. Done.